modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/atomic/GridDhtAtomicAbstractUpdateFuture.java

```diff
@@ -387,46 +393,58 @@ final void map(ClusterNode nearNode,
         GridCacheReturn ret,
         GridNearAtomicUpdateResponse updateRes,
         GridDhtAtomicCache.UpdateReplyClosure completionCb) {
-        if (F.isEmpty(mappings)) {
-            updateRes.mapping(Collections.<UUID>emptyList());
+        try (
+            TraceSurroundings ignored =
+                MTC.supportContinual(span = cctx.kernalContext().tracing().create(CACHE_API_DHT_UPDATE_FUTURE,
+                    MTC.span()));
+            TraceSurroundings ignored2 =
+                MTC.support(cctx.kernalContext().tracing().create(CACHE_API_DHT_UPDATE_MAP, span))
+        ) {
+            span.addTag("write.version", () -> Objects.toString(writeVer));
 
-            completionCb.apply(updateReq, updateRes);
+            MTC.span().addTag("mappings", () -> Objects.toString(mappings));
```

Comment:
Do mappings itself contain any useful information?
If we want to trace keys distribution between nodes, we can trace keys received in NearAtomicUpdateRequest

Reply:
We can also add tag for the moment when entry locks are acquired on primary node

Reply:
1.Fixed.
2.log points added to GridDhtAtomicCache#lockEntries and GridDhtAtomicCache#unlockEntries